### PR TITLE
Add TOTP code and AES encryption extension.

### PR DESCRIPTION
**Pull request by LIZARD-OFFICIAL**
Restored AES.js
Just made a two-factor authentication extension.
You can literally make a 2FA app in scratch now
@GarboMuffin Please merge

### CODE REVIEW

**Comment by NexusKitten** [comment]
It would help if you could separate these two extensions into separate pull requests, so we can review them individually.

**Comment by LIZARD-OFFICIAL** [comment]
> It would help if you could separate these two extensions into separate pull requests, so we can review them individually.

I can't create two forks of same repo

**Comment by NexusKitten** [comment]
> > It would help if you could separate these two extensions into separate pull requests, so we can review them individually.
> 
> I can't create two forks of same repo

Use [branches.](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository)

**Comment by CST1229** [comment]
> > > It would help if you could separate these two extensions into separate pull requests, so we can review them individually.
> > 
> > 
> > I can't create two forks of same repo
> 
> Use [branches.](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository)

Also, don't use the `master` branch.

**Comment by LIZARD-OFFICIAL** [comment]
@GarboMuffin Please add it this time.

File: extensions/LIZARD_OFFICIAL/2FA.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/* eslint-disable */
```
Comment (NexusKitten) [inline extensions/LIZARD_OFFICIAL/2FA.js:1]: Using `/* eslint-disable */` for the whole script is generally not a good idea, it's required for the library you imported but you should add an `/* eslint-enable */` afterwards.

File: extensions/LIZARD_OFFICIAL/2FA.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  "use strict";
+
+  class MFA {
+    getInfo() {
+      return {
+        id: "twofactorauth",
```
Comment (NexusKitten) [inline extensions/LIZARD_OFFICIAL/2FA.js:13]: The extension ID should contain someway to identify it's made by you. Something like "lizard2fauth" would work.

File: extensions/LIZARD_OFFICIAL/2FA.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                defaultValue: "SECRET KEY",
+              }
+          },
+          },
+        ],
+      };
+    }
+
+    gettotpcode(args) {
+      return new jsOTP.totp().getOtp(args.CODE).toString()
```
Comment (NexusKitten) [inline extensions/LIZARD_OFFICIAL/2FA.js:32]: Missing semicolon

File: extensions/LIZARD_OFFICIAL/2FA.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  class MFA {
+    getInfo() {
+      return {
+        id: "twofactorauth",
+        name: "2FA",
+        blocks: [
+          {
+            opcode: "gettotpcode",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "GET TOTP CODE [CODE]",
```
Comment (NexusKitten) [inline extensions/LIZARD_OFFICIAL/2FA.js:19]: This is more of a nitpick, but to parody blocks in Scratch it's preferred to use lowercase for your blocks unless necessary.

File: extensions/LIZARD_OFFICIAL/aes.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable */
```
Comment (NexusKitten) [inline extensions/LIZARD_OFFICIAL/aes.js:1]: Same thing here about `/* eslint-disable */`, see previous comment pertaining to it.

File: extensions/LIZARD_OFFICIAL/aes.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                defaultValue: 'KEY'
+              }
+            }
+          }
+        ]
+      };
+    }
+
+    encrypt (args) {
+      return CryptoJS.AES.encrypt(args.TEXT,args.KEY).toString()
```
Comment (NexusKitten) [inline extensions/LIZARD_OFFICIAL/aes.js:49]: Missing semicolon

File: extensions/LIZARD_OFFICIAL/aes.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+  class AESE {
+    getInfo () {
+      return {
+        id: 'AES',
```
Comment (NexusKitten) [inline extensions/LIZARD_OFFICIAL/aes.js:11]: The extension ID should contain someway to identify it's made by you. Something like "aeslizard" would work.

**Review by NexusKitten — CHANGES_REQUESTED** [review]
Some other finishing comments:
- Both extensions need to start with the metadata the other extensions do- see [CONTRIBUTING.md](https://github.com/TurboWarp/extensions/blob/master/CONTRIBUTING.md).
- Consider documentation for one or both of these extensions, as both of thier purposes are rather complex and may be confusing to other audiences.

**Comment by LIZARD-OFFICIAL** [comment]
> Some other finishing comments:
> 
> * Both extensions need to start with the metadata the other extensions do- see [CONTRIBUTING.md](https://github.com/TurboWarp/extensions/blob/master/CONTRIBUTING.md).
> * Consider documentation for one or both of these extensions, as both of thier purposes are rather complex and may be confusing to other audiences.

Did all of them, working on documentation rn.

**Comment by LIZARD-OFFICIAL** [comment]
@GarboMuffin please merge, its finished.

**Comment by LIZARD-OFFICIAL** [comment]
> > > > It would help if you could separate these two extensions into separate pull requests, so we can review them individually.
> > > 
> > > 
> > > I can't create two forks of same repo
> > 
> > 
> > Use [branches.](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository)
> 
> Also, don't use the `master` branch for pull requests.

Which one do i use then?

**Comment by LIZARD-OFFICIAL** [comment]
I'm gonna close this and open a new one with `cryptography` branch

**Pull request closed without merging**